.github/workflows/generate-bump-pages.yml

```diff
@@ -0,0 +1,73 @@
+name: Check & deploy API documentation
+
+on:
+  workflow_dispatch: # Allow manual trigger in case of quick fix
+  push:
+    branches:
+      - main
+    paths:
+      - 'openapi/**.json'
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'openapi/**.json'
+
+permissions:
+  contents: read
+
+jobs:
+  create-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          spec_mapping=$(node .github/scripts/generateSpecMapping.js)
+          echo "matrix=$spec_mapping" >> "$GITHUB_OUTPUT"
+
+  deploy-doc:
+    needs: create-matrix
+    if: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
+    name: Deploy API documentation on Bump.sh
+    strategy:
+      matrix: 
+        spec-mapping: ${{ fromJson(needs.create-matrix.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Deploy API documentation
+        uses: bump-sh/github-action@v1
+        with:
+          doc: ${{matrix.spec-mapping.doc}}
+          token: ${{secrets.BUMP_TOKEN}}
+          file: ${{matrix.spec-mapping.file}}
+          branch: ${{matrix.spec-mapping.branch}}
+  
+  api-preview:
```

Comment:
what is the experience to access the preview spec on bump once the job is completed? I am wondering if we should add a step here to add a comment to the PR with the link to the spec

Reply:
Right now, each matrix job will result in a separate preview link in the GH action logs ([example](https://github.com/rayangler/mdb-openapi/actions/runs/13598372933/job/38020034353#step:3:11)). Would you like me to file a separate ticket to investigate the best way to comment on the PR (probably without spamming a comment for each one) or is this sufficient for now?

Reply:
> Right now, each matrix job will result in a separate preview link in the GH action logs

SGTM. Feel free to introduce it in separate PR and spec out how to get access to preview urls.